poetry/semver/constraint_utils.py

```diff
@@ -0,0 +1,48 @@
+"""
+Constraint utilities
+
+The focus of this module is any constraint support, but
+PEP-0440 support trumps everything else.
+
+WRT, SEM-VER, PEP-0440 is partially but not fully
+compatible with all the SEM-VER specs.
+
+https://legacy.python.org/dev/peps/pep-0440/#semantic-versioning
+> Semantic versions containing a hyphen (pre-releases - clause 10) or
+> a plus sign (builds - clause 11) are not compatible with this PEP
+> and are not permitted in the public version field.
+"""
+
+from typing import List
+from typing import Optional
+
+from poetry.semver import parse_single_constraint
+
+
+def is_constraint(value):  # type: (str) -> bool
+    """Check that a string is a single constraint
+
+    :param value: any string
+    :return: True if value is a constraint
+    """
+    try:
+        if parse_single_constraint(value):
+            return True
+        else:
+            return False
+    except (TypeError, ValueError):
+        return False
+
+
+def sorted_constraints(
+    values, reverse=False
+):  # type: (List[str], Optional[bool]) -> List[str]
```

Comment:
are optional booleans a thing? there may be rare cases where it should be employed, but not here.

Reply:
huh?  `reverse=False` is an optional boolean arg, what else can it be?  In general, in other places, I'm using type annotation for this like the following and the system just takes care of the optional detail automatically; I'm not so familiar with the `# type` conventions (so what exactly should this be please?).

`values: List[str], reverse: bool = False`